sql/filesort.cc

```diff
@@ -226,7 +226,7 @@ ha_rows filesort(THD *thd, TABLE *table, SORT_FIELD *sortorder, uint s_length,
       }
       if ((table_sort.sort_keys=
            make_char_array(table_sort.sort_keys,
-                           param.keys, param.rec_length, MYF(0))))
+                           param.keys, param.rec_length, MYF(MY_ZEROFILL))))
```

Comment:
A bit more info here: this isn't results in taking any incorrect paths - it's only about copying random memory garbage around, and I think we also write it to the disk sometimes.

If we say that this is ok to do, I could mark these allocations as initialized instead of really initializing them.

Reply:
Unless there is a noticeable performance hit, better to initialize. BTW, there used to be Valgrind errors on "sycall pointing to uninit bytes" which match the description of I/O on uninit data here.